cinder/volume/drivers/vmware/vmdk.py

```diff
@@ -489,8 +490,24 @@ def _get_volume_stats(self):
                 summary = datastore["summary"]
 
                 pool_state = 'down'
+                pool_down_reason = 'Datastore not usable'
                 if self.ds_sel.is_datastore_usable(summary):
                     pool_state = 'up'
+                    pool_down_reason = 'up'
+
+                # make sure the datastore isn't in red flag alert!
+                if 'alarms' in datastore and datastore['alarms']:
+                    for alarm in datastore['alarms']:
+                        if (alarm[0]['overallStatus'] == 'red' or
+                                alarm[0]['overallStatus'] == 'yellow'):
+                            alarm_info = self.volumeops.get_datastore_alarm(
+                                alarm[0]['alarm']
```

Comment:
alarm[0] is the alarm state that as the status.

Reply:
Why is it' `alarm[0]` though and not `alarm`? Why is `alarm` a list or tuple of things and of what things? From the code and from trying to reproduce this locally in a shell, I cannot see how this would be a list and would need some more explanation, please.

Reply:
ok I see the issue now.  working on a fix for it.